### PR TITLE
ci: Integration test automation workflow

### LINKED ISSUE #741: Add GitHub Actions workflow to run the integration test suite
[Refer the `ai.backend.test` source directory.](https://github.com/lablup/backend.ai/tree/main/src/ai/backend/test)

Let's run it [periodically](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) and [manually requested](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch).

### PR DESCRIPTION
**Pull request by Yaminyam**
close #741 
Add `script -q -e -c "bash {0}"` shell setting for tty use by `-i` option of `docker exec` according to the environment in github actions

Periodic execution using crontab
Executed by manual trigger using workflow_dispatch.

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: integration-test
+
+on:
+  schedule:
+    - cron: '0 0 * * 6'
```
Comment (fregataa) [inline .github/workflows/integration.yml:5]: this test runs every saturday 00:00 ?

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - name: Apply cache for Pants
+      uses: actions/cache@v3
+      id: cache
+      with:
+        # pants-specific cache
+        path: |
+          ~/.cache/pants/setup
+          ~/.cache/pants/lmdb_store
+          ~/.cache/pants/named_caches
+        key: ${{ runner.os }}-${{ hashFiles('pants*.toml', '**/*.lock') }}-
```
Comment (fregataa) [inline .github/workflows/integration.yml:27]: Every test reuses this cache?
And how long this cache remain?

**Review by fregataa — COMMENTED** [review]
I left some comments

File: .github/workflows/integration.yml (hunk 1 above)
Comment (Yaminyam) [reply .github/workflows/integration.yml:5]: That's right. This test runs every Saturday on the hour.
It seems to run at 10:00 Korean time.

File: .github/workflows/integration.yml (hunk 2 above)
Comment (Yaminyam) [reply .github/workflows/integration.yml:27]: https://github.com/lablup/backend.ai/actions/caches
Recently updated github to check caches in .

I don't know exactly how much cache is maintained, but I checked until the cache was over a month old.

Comment (Yaminyam) [reply .github/workflows/integration.yml:27]: https://github.com/actions/cache
Caches that have not been accessed within the past week are deleted, and if the 10GB capacity is full, the oldest cache is deleted.

**Review by achimnol — APPROVED** [review]
LGTM!
Let's run it first and fix any issues afterwards.

**Pull request merged**